### PR TITLE
Update WI related docs after recent 0.15 release cut 

### LINKED ISSUE #1143: Update WI related docs after recent 0.15 release cut
**Problem**
Update WI related docs after recent 0.15 release cut, as we remove gsa and add ksa in v1beta1

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
Update WI related docs after recent 0.15 release cut

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by grac3gao**
Fixes #1143 

## Proposed Changes

- Update data plane setups
- Update examples
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: docs/examples/topic/README.md [hunk 1]
```diff
@@ -25,9 +25,8 @@ construct used by higher-level objects, such as `Channel`.
 
    1. If you are in GKE and using
       [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity),
-      update `googleServiceAccount` with the Pub/Sub enabled service account you
-      created in
-      [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md).
+      update `serviceAccountName` with the Kubernetes service account you created in [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md), 
+      which is bound to the Pub/Sub enabled service account.
```
Comment (Harwayne) [inline docs/examples/topic/README.md:29]:
```suggestion
      which is bound to the Pub/Sub enabled Google service account.
```

File: docs/examples/cloudbuildsource/README.md [hunk 2]
```diff
@@ -31,9 +31,8 @@ build completes.
 
    1. If you are in GKE and using
       [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity),
-      update `googleServiceAccount` with the Pub/Sub enabled service account you
-      created in
-      [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md).
+      update `serviceAccountName` with the Kubernetes service account you created in [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md), 
+      which is bound to the Pub/Sub enabled google service account.
```
Comment (Harwayne) [inline docs/examples/cloudbuildsource/README.md:35]:
```suggestion
      which is bound to the Pub/Sub enabled Google service account.
```

File: docs/examples/cloudschedulersource/README.md [hunk 3]
```diff
@@ -35,9 +35,8 @@ scheduled events from
 
    1. If you are in GKE and using
       [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity),
-      update `googleServiceAccount` with the Pub/Sub enabled service account you
-      created in
-      [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md).
+      update `serviceAccountName` with the Kubernetes service account you created in [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md), 
+      which is bound to the Pub/Sub enabled google service account.
```
Comment (Harwayne) [inline docs/examples/cloudschedulersource/README.md:39]:
```suggestion
      which is bound to the Pub/Sub enabled Google service account.
```

File: docs/examples/cloudstoragesource/README.md [hunk 4]
```diff
@@ -77,14 +77,13 @@ Object Notifications for when a new object is added to Google Cloud Storage
...
+1. Update `serviceAccountName` / `secret` in the
    [`cloudstoragesource.yaml`](cloudstoragesource.yaml)
 
    1. If you are in GKE and using
       [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity),
-      update `googleServiceAccount` with the Pub/Sub enabled service account you
-      created in
-      [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md).
+      update `serviceAccountName` with the Kubernetes service account you created in [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md), 
+      which is bound to the Pub/Sub enabled google service account.
```
Comment (Harwayne) [inline docs/examples/cloudstoragesource/README.md:86]:
```suggestion
      which is bound to the Pub/Sub enabled Google service account.
```

File: docs/examples/pullsubscription/README.md [hunk 5]
```diff
@@ -29,9 +29,8 @@ does so using a Pull format.
 
    1. If you are in GKE and using
       [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity),
-      update `googleServiceAccount` with the Pub/Sub enabled service account you
-      created in
-      [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md).
+      update `serviceAccountName` with the Kubernetes service account you created in [Create a Pub/Sub enabled Service Account](../../install/pubsub-service-account.md), 
+      which is bound to the Pub/Sub enabled google service account.
```
Comment (Harwayne) [inline docs/examples/pullsubscription/README.md:33]:
```suggestion
      which is bound to the Pub/Sub enabled Google service account.
```

File: docs/install/authentication-mechanisms-gcp.md [hunk 6]
```diff
@@ -76,25 +75,38 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
       **_NOTE_**: These commands may take a long time to finish. Check
       [Enable Workload Identity on an existing cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_on_an_existing_cluster)
       for more information.
-
-1. Bind the Kubernetes Service Account `controller` with Google Cloud Service
+1. Create a Kubernetes Service Account `ksa-name` in namespace your resources will reside.
```
Comment (Harwayne) [inline docs/install/authentication-mechanisms-gcp.md:78]:
```suggestion
1. Create a Kubernetes Service Account `ksa-name` in the namespace your resources will reside. 
```

File: docs/install/authentication-mechanisms-gcp.md [hunk 7]
```diff
@@ -76,25 +75,38 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
...
    --role roles/iam.workloadIdentityUser \
-   --member $MEMBER cloud-run-events@$PROJECT_ID.iam.gserviceaccount.com
+   --member $MEMBER gsa-name@$PROJECT_ID.iam.gserviceaccount.com
    ```
+   
+   If you are configuraing Authentication Mechanism for the Control Plane, you can replace `ksa-namespace` with `cloud-run-events`, 
+   `ksa-name` with `controller`, and `gsa-name` with `cloud-run-events`
 
-1. Add annotation to Kubernetes Service Account `controller`.
+1. Add annotation to Kubernetes Service Account `ksa-name`.
```
Comment (Harwayne) [inline docs/install/authentication-mechanisms-gcp.md:100]:
```suggestion
1. Annotate the Kubernetes Service Account `ksa-name`.
```

File: docs/install/authentication-mechanisms-gcp.md [hunk 8]
```diff
@@ -76,25 +75,38 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
...
 
    ```shell
-   kubectl annotate serviceaccount controller iam.gke.io/gcp-service-account=cloud-run-events@$PROJECT_ID.iam.gserviceaccount.com \
-   --namespace cloud-run-events
+   kubectl annotate serviceaccount ksa-name iam.gke.io/gcp-service-account=gsa-name@$PROJECT_ID.iam.gserviceaccount.com \
+   --namespace ksa-namespace
    ```
 
+   If you are configuraing Authentication Mechanism for the Control Plane, you can replace `ksa-namespace` with `cloud-run-events`,
```
Comment (Harwayne) [inline docs/install/authentication-mechanisms-gcp.md:107]:
```suggestion
   If you are configuring Authentication Mechanism for the Control Plane, you can replace `ksa-namespace` with `cloud-run-events`, 
```

File: docs/install/pubsub-service-account.md [hunk 9]
```diff
@@ -55,65 +55,22 @@ It is the recommended way to access Google Cloud services from within GKE due to
...
-
-   For every namespace, the Kubernetes Service Account and the Google Cloud
-   Service Account will have a one to one mapping. If multiple source instances
-   use the same Google Cloud Service Account, they will use the same Kubernetes
-   Service Account as well, and the Kubernetes Service Account will have
-   multiple `OwnerReference`s. If all sources using a Kubernetes Service Account
-   are deleted then that Kubernetes Service Account, and its corresponding
-   Google Cloud Service account, will be deleted.
+**_Note:_**  If you install 
+the Knative-GCP Constructs with v0.13.0 or older release, please use option 2.
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:60]:
```suggestion
the Knative-GCP Constructs with v0.14.0 or older release, please use option 2.
```

To match authentication-mechanisms-gcp.md.

File: docs/install/pubsub-service-account.md [hunk 10]
```diff
@@ -55,65 +55,22 @@ It is the recommended way to access Google Cloud services from within GKE due to
...
+            
+**_Note:_**  `spec.googleServiceAccount` in v0.14.0 is deprecated for security implications. 
+It has not been promoted to v1beta1 and is expected to be removed from v1alpha1 in the v0.16.0 release.
+Instead, `spec.serviceAccountName` is introduced for Workload Identity in v0.15.0, whose input is a Kubernetes Service Account.
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:64]:
```suggestion
Instead, `spec.serviceAccountName` has been introduced for Workload Identity in v0.15.0, whose value is a Kubernetes Service Account.
```

**Comment by Harwayne** [comment]
I'm not sure why this is a problem, but it looks like Prow is complaining about:

```
docs/install/install-gcp-broker.md
	ERROR	https://knative.dev/docs/eventing/broker-trigger/#broker
		Not Found (HTTP error 404)
	ERROR	https://knative.dev/docs/eventing/broker-trigger/#trigger
		Not Found (HTTP error 404)
	ERROR	https://knative.dev/docs/eventing/channel-based-broker/
		Not Found (HTTP error 404)
```

**Comment by grac3gao** [comment]
@Harwayne looks like some links in broker doc are no longer existing. I did a small update to include new links...

**Pull request merged**